src/cudamatrix/cu-kernels.cu

```diff
@@ -741,6 +741,16 @@ static void _set_bias_params(Real* v, const Real* a, Real param_1, Real param_2,
   }
 }
 
+template<typename Real, typename OtherReal>
+__global__
+static void _copy_from_vec(int n, const Real* x, int incx, OtherReal* y, int incy) {
+  int32_cuda i = blockIdx.x * blockDim.x + threadIdx.x;
+   // if (blockIdx.y > 0) return;
+
+  if (i < n) {
+    y[1+(i-1)*incy] = (OtherReal) x[1+(i-1)*incx];
```

Comment:
cuBLAS API has both 0-based and 1-based indexing. For fortran code ported to C, 1-based indexing can be used for easier loop transfer. For C and C++ code, we use 0-based indexing.
Other point is that "i" is defined as blockIdx.x * blockDim.x + threadIdx.x and all blockIdx, threadIdx and blockDim ranges [0 ~ m) and i starts from 0 and you can not define (i-1)*incy even with 1-based indexing!!!!